### PR TITLE
Develop to Master (Patch & CI Changes)

### PR DESCRIPTION
**Pull request by mrz1836**
- Bumped deps
- Updated the Github Actions
- Updated makefiles
- Added more test cases
- Patching issue: `not signing multiple inputs from same address`

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -11,14 +11,13 @@ on:
     branches:
       - "*"
   schedule:
-    - cron: '* 22 * * *'
+    - cron: '1 4 * * *'
```
Comment (jadwahab) [inline .github/workflows/run-tests.yml:14]: you're running this once every hour at 04:01?

File: tx_test.go [hunk 2]
```diff
@@ -485,6 +485,135 @@ func TestTx_Change(t *testing.T) {
...
+		wif, err = bsvutil.DecodeWIF("L3MhnEn1pLWcggeYLk9jdkvA2wUK1iWwwrGkBbgQRqv6HPCdRxuw")
+		assert.NoError(t, err)
+		assert.NotNil(t, wif)
+
+		_, err = tx.SignAuto(&bt.InternalSigner{PrivateKey: wif.PrivKey, SigHashFlag: 0})
+		assert.NoError(t, err)
+
+		assert.Equal(t, "01000000010b94a1ef0fb352aa2adc54207ce47ba55d5a1c1609afda58fe9520e472299107000000006b483045022100fd07316603e9abf393e695192e8ce1e7f808d2735cc57039109a2210ad32d9a7022000e301e2a988b23ab3872b041df8b6eb0315238e0918944cbaf8b6abdde75cac412102c8803fdd437d902f08e3c2344cb33065c99d7c99982018ff9f7219c3dd352ff0ffffffff023b420f00000000001976a914b6aa34534d2b11e66b438c7525f819aee01e397c88acc0c62d00000000001976a914b6aa34534d2b11e66b438c7525f819aee01e397c88ac00000000", tx.ToString())
+
+		// todo: expected the pay-to inputs to change based on the fee :P
```
Comment (jadwahab) [inline tx_test.go:580]: not sure what exactly what you mean by this. only the third output should change based on what fees you supply

**Review by jadwahab — APPROVED** [review]

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (mrz1836) [reply .github/workflows/run-tests.yml:14]: No, only once a day ;-)

File: tx_test.go (hunk 2 above)
Comment (mrz1836) [reply tx_test.go:580]: I probably had the wrong expectation... expected magic to happen and fees to automatically be deducted... :P

**Pull request merged**